### PR TITLE
feat: modifiy ReceiptWithBloom and associated impls to use with Reth

### PR DESCRIPTION
**Pull request by royvardhan**
Unblocks: https://github.com/paradigmxyz/reth/issues/12701

### CODE REVIEW

**Comment by royvardhan** [comment]
@mattsse As discussed, before updating `ReceiptWithBloom` type, I have updated TxReceipt trait and also changed the associated impls. Wdyt?

File: crates/rpc-types-eth/src/transaction/receipt.rs [hunk 1]
```diff
@@ -95,7 +95,14 @@ impl TransactionReceipt {
...
     pub const fn transaction_type(&self) -> TxType {
-        self.inner.tx_type()
+        match &self.inner {
+            ReceiptEnvelope::Eip1559(_) => TxType::Eip1559,
+            ReceiptEnvelope::Eip2930(_) => TxType::Eip2930,
+            ReceiptEnvelope::Eip4844(_) => TxType::Eip4844,
+            ReceiptEnvelope::Eip7702(_) => TxType::Eip7702,
+            ReceiptEnvelope::Legacy(_) => TxType::Legacy,
+            _ => TxType::Legacy,
+        }
```
Comment (mattsse) [inline crates/rpc-types-eth/src/transaction/receipt.rs:105]: why can we no longer use the function?

**Review by mattsse — COMMENTED** [review]
looks good so far,

what's missing?

cc @klkvr

**Comment by royvardhan** [comment]
@mattsse I think we are good.

**Review by mattsse — APPROVED** [review]
lgtm, 
this should be simple to port to op-alloy, I believe

**Comment by royvardhan** [comment]
> lgtm, 
> 
> this should be simple to port to op-alloy, I believe

On it.

**Pull request merged**